### PR TITLE
[bug, feat] Replicate Fig 3 in Frajka-Williams et al. 2011

### PR DESCRIPTION
**Pull request by eleanorfrajka**
- Add a new sample dataset in demo.ipynb, 
- Tool for w: called `ramsey_binavg()` which takes the w data and bins it in depth
- Plotting tool for w: to plot up-down bias in w based on ramsey_binavg().

Also
- Fixed a plotting error in plot_vertical_speeds_with_histograms.  Removed ax.twinx which allowed the w data to look offset because there were two different y-axes

### CODE REVIEW

**Comment by callumrollo** [comment]
Nice addition! The docs are failing because it's running these new functions on the default dataset, `sea045_20230530T0832_delayed.nc`, which doesn't have the variable VERT_CURR_MODEL. Can you send me the new dataset `sg014_20040924T182454_delayed.nc`? I'll upload it to the server then you can use it in the demo notebook for these functions like:

```python
ds_sg014 = fetchers.load_sample_dataset(dataset_name="sg014_20040924T182454_delayed.nc") # Full Seaglider mission in the Labrador Sea

tools.plot_vertical_speeds_with_histograms(ds_sg014, start_prof, end_prof)
```

etc.

File: glidertest/tools.py [hunk 1]
```diff
@@ -1100,3 +1097,161 @@ def plot_vertical_speeds_with_histograms(ds, start_prof=None, end_prof=None):
...
+        CIlimits = .95 # Could be passed as a variable. 0.95 for 95% confidence intervals
+        if len(ifind):
+            meanw[zdo] = np.nanmean(ww[ifind])
+
+            # Confidence intervals
+            # Number of data points used in the mean at this depth (zgrid[zdo])
+            NNz[zdo] = len(ifind)
+            if NNz[zdo] > 1:
+                se = np.nanstd(ww[ifind]) / np.sqrt(NNz[zdo])  # Standard error
+                ci = se * t.ppf((1 + CIlimits) / 2, NNz[zdo] - 1)  # Confidence interval based on CIlimits
```
Comment (callumrollo) [inline glidertest/tools.py:1168]: t.ppf doesn't seem to be defined in the committed code. Does it need to be imported/calculated somewhere?

Comment (eleanorfrajka) [reply glidertest/tools.py]: Yep, sorry, should be stats.t.ppf.   My version had from scipy.stats import t

**Comment by callumrollo** [comment]
I think it's just that in the new notebook cells you need to use `ds_sg015` rather than `ds` for all these vertical velocity functions like `plot_vertical_speeds_with_histograms` and `plot_combined_velocity_profiles`

**Comment by eleanorfrajka** [comment]
> I think it's just that in the new notebook cells you need to use `ds_sg015` rather than `ds` for all these vertical velocity functions like `plot_vertical_speeds_with_histograms` and `plot_combined_velocity_profiles`

Yes!  Think I got it, but I did it with ds_sg014 (which is the bigger file).

**Comment by eleanorfrajka** [comment]
Note: It does run on the smaller sample sg015 file, but the output is a little nuts.
51c841e7-eaff-4b80-a5aa-30805b8b5bd5 

Compared to (for sg014): 
f97ef1c2-d09a-46c6-aa8c-a4f2361d9627

I'm not sure what the preferred plan is for demo notebooks.  Could be an option to make a more static version showing what the outputs could look like if you download the larger file - seems mean to make people download a big file to run the test code, but also a little misleading to show what happens to the diagnostic when you only have a few profiles to average.

**Comment by callumrollo** [comment]
Yeah we can work with something like that for sure, generating some static images for docs but keeping the standard notebook to the smaller example datasets. Perhaps Chiara could work her magic cutting down this sea014 file to a usefully small set of profiles?

At some point we'll also likely break the notebook up into several examples, so it may happen naturally that vertical velocities become their own seperate demo

**Comment by MOchiara** [comment]
Yeah sure! I can chop sea014 so we have a decent example but smaller file

**Pull request merged**